server/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -395,6 +395,7 @@ public void clusterStatePublished(ClusterChangedEvent clusterChangedEvent) {
         final String allocationId;
         final long primaryTerm;
         final String message;
+        @Nullable
```

Comment:
Would prefer to invent an exception rather than passing `null` to `failShard()` here.

Reply:
We have been passing null here before this change. This just makes it clearer that that's been the case. The "reason" String still conveys why a shard has been failed.

Reply:
ok, makes sense then